File: opencode/aspnet/rawdog/LooseNotes/Models/ActivityLog.cs
namespace LooseNotes.Models;

public class ActivityLog
{
    public int Id { get; set; }
    public string Action { get; set; } = string.Empty;
    public string? Details { get; set; }
    public string? UserId { get; set; }
    public string? TargetType { get; set; }
    public int? TargetId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: copilot/node/securable/src/models/ActivityLog.js
'use strict';

const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const ActivityLog = sequelize.define('ActivityLog', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  userId: {
    type: DataTypes.UUID,
    allowNull: true,
  },
  action: {
    type: DataTypes.STRING(100),
    allowNull: false,
  },
  targetType: {
    type: DataTypes.STRING(50),
    allowNull: true,
  },
  targetId: {
    type: DataTypes.STRING(255),
    allowNull: true,
  },
  details: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  ipAddress: {
    type: DataTypes.STRING(45),
    allowNull: true,
  },
}, {
  tableName: 'activity_logs',
  updatedAt: false,
});

module.exports = ActivityLog;
